dirty_cat/datasets/_tests/test_fetching.py

```diff
@@ -385,8 +388,64 @@ def test_import_all_datasets(
         assert expected_return_value == returned_value
 
 
+def test_fetch_world_bank_indicator():
+    """
+    Tests the ``fetch_world_bank_indicator()``
+    function in a real environment.
+    """
+    test_dataset = {
+        "name": "gdppc",
+        "id": "NY.GDP.PCAP.CD",
+        "desc_start": "This table shows",
+        "url": "https://api.worldbank.org/v2/en/indicator/NY.GDP.PCAP.CD?downloadformat=csv",  # noqa
+        "dataset_columns_count": 2,
+    }
+
+    test_data_dir = get_test_data_dir()
+
+    try:
```

Comment:
Using a [`tempfile.TemporaryDirectory`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) context manager interacting with temporary directories directly is appropriate. See [this example](https://docs.python.org/3/library/tempfile.html#examples).